compose-lint-checks/src/main/java/slack/lint/compose/M2ApiDetector.kt

```diff
@@ -0,0 +1,70 @@
+// Copyright (C) 2022 Salesforce, Inc.
+// SPDX-License-Identifier: Apache-2.0
+package slack.lint.compose
+
+import com.android.tools.lint.client.api.UElementHandler
+import com.android.tools.lint.detector.api.Category.Companion.CORRECTNESS
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.JavaContext
+import com.android.tools.lint.detector.api.Severity.IGNORE
+import com.android.tools.lint.detector.api.SourceCodeScanner
+import com.android.tools.lint.detector.api.TextFormat
+import org.jetbrains.uast.UCallExpression
+import org.jetbrains.uast.UElement
+import org.jetbrains.uast.UImportStatement
+import org.jetbrains.uast.UQualifiedReferenceExpression
+import org.jetbrains.uast.UResolvable
+import slack.lint.compose.util.Priorities.NORMAL
+import slack.lint.compose.util.sourceImplementation
+
+internal class M2ApiDetector : Detector(), SourceCodeScanner {
+
+  override fun getApplicableUastTypes() =
+    listOf<Class<out UElement>>(
+      UCallExpression::class.java,
+      UImportStatement::class.java,
+      UQualifiedReferenceExpression::class.java,
+    )
+
+  override fun createUastHandler(context: JavaContext) =
+    object : UElementHandler() {
+      override fun visitCallExpression(node: UCallExpression) = checkNode(node)
+
+      override fun visitImportStatement(node: UImportStatement) = checkNode(node)
+
+      override fun visitQualifiedReferenceExpression(node: UQualifiedReferenceExpression) =
+        checkNode(node)
+
+      private fun checkNode(node: UResolvable) {
+        val resolved = node.resolve() ?: return
+        val packageName = context.evaluator.getPackage(resolved)?.qualifiedName ?: return
+        if (packageName == M2Package) {
+          context.report(
+            issue = ISSUE,
+            location = context.getLocation(node),
+            message = ISSUE.getExplanation(TextFormat.TEXT),
+          )
+        }
+      }
+    }
+
+  companion object {
+    private const val M2Package = "androidx.compose.material"
+
+    val ISSUE =
+      Issue.create(
+        id = "ComposeM2Api",
+        briefDescription = "Using a Compose M2 API is not recommended",
+        explanation =
+          """
+          Compose Material 2 (M2) is succeeded by Material 3 (M3). Please use M3 APIs.
+          See https://slackhq.github.io/compose-lints/rules/#use-material-3 for more information.
+        """,
+        category = CORRECTNESS,
+        priority = NORMAL,
+        severity = IGNORE,
```

Comment:
informational? I think it's worth highlighting even if devs don't want to make it an error.

Reply:
I think I generally agree. Do you know if M2 is ever going to be fully deprecated?

Reply:
Ah good point. We need to think about teams who are using M2 and don't want to migrate yet. We should probably add some docs to copy-pasta for changing the severity level?

Reply:
[d1f8079](https://github.com/slackhq/compose-lints/pull/75/commits/d1f8079a1c439d9c7986e64a1f52a2df20486a35)